public/app/core/internationalization/index.tsx

```diff
@@ -29,11 +33,15 @@ export function initializeI18n(language: string) {
     'Do not manually edit this file, or update these source phrases in Crowdin. The source of truth for English strings are in the code source'
   );
 
-  return i18n
+  let init = i18n;
+  if (!validLanguage) {
+    init = init.use(LanguageDetector);
```

Comment:
I think it might be best to customise the options here to only use the navigator detector.

I spent a bunch of time while reviewing this PR trying to understand why my browser preference wasn't being reflected any more, but it was because by default this 'caches' the value to local storage, causing browser preference to no longer be obeyed. Strange design decision by the library...